src/System.ServiceModel.Primitives/tests/ServiceModel/DuplexChannelFactoryTest.cs

```diff
@@ -0,0 +1,384 @@
+    [Fact]
+    public static void CreateChannel_Of_IDuplexChannel_Using_BasicHttpBinding()
+    {
+        ChannelFactory<IWcfDuplexService> factory = null;
+        ChannelFactory<IWcfDuplexService> factory2 = null;
+        IWcfDuplexService channel = null;
+        IWcfDuplexService channel2 = null;
+
+        WcfDuplexServiceCallback callback = new WcfDuplexServiceCallback();
+
+        try
+        {
+            BasicHttpBinding binding = new BasicHttpBinding();
+            EndpointAddress endpointAddress = new EndpointAddress(BaseAddress.HttpBaseAddress);
+
+            // Create the channel factory for the duplex message exchange pattern.
+            factory = new DuplexChannelFactory<IWcfDuplexService>(callback, binding, endpointAddress);
+            factory2 = new DuplexChannelFactory<IWcfDuplexService>(callback, binding, endpointAddress);
+
+            Assert.Equal(CommunicationState.Created, factory.State);
+            Assert.Equal(CommunicationState.Created, factory2.State);
+
+            var exception = Assert.Throws<InvalidOperationException>(() =>
+            {
+                // Create the channel.
+                channel = factory.CreateChannel();
+                Assert.True(typeof(IWcfDuplexService).GetTypeInfo().IsAssignableFrom(channel.GetType().GetTypeInfo()),
+                    String.Format("Channel type '{0}' was not assignable to '{1}'", channel.GetType(), typeof(IDuplexChannel)));
+            });
+
+            // BasicHttpBinding doesn't support duplex, hence InvalidOperationException
+            // the state of the channel factory should be "Faulted" 
+            Assert.Contains("BasicHttpBinding", exception.Message); 
+            Assert.Equal(CommunicationState.Faulted, factory.State);
+            
+            var exception2 = Assert.Throws<InvalidOperationException>(() =>
+            {
+                channel2 = factory2.CreateChannel();
+                Assert.True(typeof(IWcfDuplexService).GetTypeInfo().IsAssignableFrom(channel2.GetType().GetTypeInfo()),
+                    String.Format("Channel type '{0}' was not assignable to '{1}'", channel2.GetType(), typeof(IDuplexChannel)));
+            });
+
+            // BasicHttpBinding doesn't support duplex, hence InvalidOperationException
+            // the state of the channel factory should be "Faulted" 
+            Assert.Contains("BasicHttpBinding", exception.Message); 
+            Assert.Equal(CommunicationState.Faulted, factory2.State);
+
+        }
+        finally
+        {
+            if (factory != null)
+            {
+                factory.Abort();
+                Assert.Equal(CommunicationState.Closed, factory.State);
+            }
+            if (factory2 != null)
+            {
+                factory2.Abort();
+                Assert.Equal(CommunicationState.Closed, factory2.State);
+            }
+        }
+    }
+
+    [Fact]
+    public static void CreateChannel_Of_IDuplexChannel_Using_NetTcpBinding()
+    {
```

Comment:
This test is really validating that 2 calls to CreateChannel creates unique channel instances.  I'd rename it to reflect that.